components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -2222,6 +2172,92 @@ private String handlePreConsent(OAuth2Parameters oauth2Params, AuthenticatedUser
         return additionalQueryParam;
     }
 
+    /**
+     * Filter requested claims based on OIDC claims and return the claims which includes in OIDC.
+     *
+     * @param claimsForApproval         Consent required claims.
+     * @param oauth2Params              OAuth parameters.
+     * @param spTenantDomain            Tenant domain.
+     * @return                          Requested OIDC claim list.
+     * @throws RequestObjectException   If an error occurred while getting essential claims for the session data key.
+     * @throws ClaimMetadataException   If an error occurred while getting claim mappings.
+     */
+    private List<ClaimMetaData> getRequestedOidcClaimsList(ConsentClaimsData claimsForApproval,
+                                                           OAuth2Parameters oauth2Params, String spTenantDomain)
+            throws RequestObjectException, ClaimMetadataException {
+
+        List<ClaimMetaData> requestedOidcClaimsList = new ArrayList<>();
+        List<String> localClaimsOfOidcClaims = new ArrayList<>();
+        List<String> localClaimsOfEssentialClaims = new ArrayList<>();
+
+        // Get the claims uri list of all the requested scopes. Eg:- country, email.
+        List<String> claimListOfScopes = openIDConnectClaimFilter.getClaimsFilteredByOIDCScopes(oauth2Params.
+                getScopes(), spTenantDomain);
+
+        // Get the requested claims came through request object.
+        List<RequestedClaim> requestedClaimsOfIdToken = EndpointUtil.getRequestObjectService()
+                .getRequestedClaimsForSessionDataKey(oauth2Params.getSessionDataKey(), false);
+
+        List<RequestedClaim> requestedClaimsOfUserInfo = EndpointUtil.getRequestObjectService()
+                .getRequestedClaimsForSessionDataKey(oauth2Params.getSessionDataKey(), true);
+
+        List<String> essentialRequestedClaims = new ArrayList<>();
+
+        // Get the list of id token's essential claims.
+        for (RequestedClaim requestedClaim : requestedClaimsOfIdToken) {
+            if (requestedClaim.isEssential()) {
+                essentialRequestedClaims.add(requestedClaim.getName());
+            }
+        }
+
+        // Get the list of user info's essential claims.
+        for (RequestedClaim requestedClaim : requestedClaimsOfUserInfo) {
+            if (requestedClaim.isEssential()) {
+                essentialRequestedClaims.add(requestedClaim.getName());
+            }
+        }
+
+        if (CollectionUtils.isNotEmpty(claimListOfScopes)) {
+            // Get the external claims relevant to all oidc scope claims and essential claims.
+            Set<ExternalClaim> externalClaimSetOfOidcClaims = ClaimMetadataHandler.getInstance()
+                    .getMappingsFromOtherDialectToCarbon
+                            (OIDC_DIALECT, new HashSet<String>(claimListOfScopes), spTenantDomain);
```

Comment:
Seems line breaks are not in the proper format. please check other places as well

Reply:
Fixed